gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/BytecodeUtil.java

```diff
@@ -43,4 +47,16 @@ private BytecodeUtil() {}
                         s -> s.getOperator().equals(TraversalSource.Symbols.withStrategies) && clazz.isAssignableFrom(s.getArguments()[0].getClass())),
                 os -> (A) os.getArguments()[0]);
     }
+
+    /**
+     * Parses {@link Bytecode} to find {@link Step} objects in the step instructions.
+     * @return
+     */
+    public static Iterator<Step> findSteps(final Bytecode bytecode, final Class<?> clazz) {
```

Comment:
I sense that we will do more with `Bytecode` analysis so this class will grow in functionality. Perhaps this PR is not the time to think it all through, but I'd offer that I think that we would want to avoid doing a `Translator` operation here and try to work on pure bytecode when doing these sorts of operations. I see the main use for this method in this PR is to detect lambdas. If that is the case, then I would instead write a `hasLambda()` function here and have it simply dig through the bytecode to look for `Lambda` objects.

Reply:
OK, I will do that. Note that the bytecode analysis of lambdas depends on magic strings that will come from the JavaTranslator anyway. So, this implies moving the magic strings to ByteUtil and also touching JavaTranslator (for the better!)

Reply:
> Note that the bytecode analysis of lambdas depends on magic strings

I'm not sure what you mean by that. I would expect the `Bytecode` object to contain an actual `org.apache.tinkerpop.gremlin.util.function.Lambda` object within it. You just need to detect that, no?

Reply:
That was an excellent pointer and brought me to BytecodeHelper.getLambdaLanguage(bytecode).isPresent(). This means that I will not touch the BytecodeUtil code any more. 
You may want to merge the BytecodeUtil and  BytecodeHelper classes in a separate PR, so that you can properly deprecate one in 3.4.x and get things right for 3.5.0 (assuming there is no valid reason for these classes to both exist). In a large codebase it is easy to overlook the other if you arrive at one via your IDE goTo.

Reply:
yeah - thanks for pointing that out. i shouldnt have added `BytecodeUtil`....

Reply:
I've implemented TINKERPOP-2462 on 7325846aeecd8ae86f3df1565def64df1bc10b13 - going to remove `BytecodeUtil` completely

Reply:
Removed here: 6e47f532933b8a4d0ec25b8aeea8cd157584c358 - when you rebase it will be gone.

Reply:
Great! Over here, documentation is becoming completer.